src/VisualStudio.Roslyn.SDK/ComponentDebugger/Roslyn.ComponentDebugger.csproj

```diff
@@ -8,13 +8,33 @@
     <Nullable>enable</Nullable>
     <UseWpf>true</UseWpf>
   </PropertyGroup>
+  <ItemGroup>
+    <Page Remove="ComponentDebuggerLaunchProfile.xaml" />
+    <XamlPropertyRule Include="ComponentDebuggerLaunchProfile.xaml">
+      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
+      <Generator>MSBuild:GenerateRuleSourceFromXaml</Generator>
+      <SubType>Designer</SubType>
+      <DataAccess>None</DataAccess>
+      <RuleInjection>None</RuleInjection>
+    </XamlPropertyRule>
+  </ItemGroup>
+  <Target Name="EmitAssemblyVersion" BeforeTargets="CoreCompile">
+    <WriteLinesToFile File="$(BaseIntermediateOutputPath)\AssemblyVersion.g.cs" 
```

Comment:
🤢This is unpleasant, but required for the rules. It seems overkill to make it a generator, but I'm considering writing one to handle the whole rule injection which would remove a lot of the ceremony needed here.

Reply:
I would prefer a source generator here. Nerdbank.GitVersioning might allow this to be automated without a source generator or project file changes.

@AArnott does NB.GV support setting the AssemblyVersion via the project file instead of the normal approach of automatically determining it from the repo history?

Reply:
No. NB.GV will steamroll any `AssemblyVersion` property you set with the discovered version. 
If you wrote build authoring to execute your target in between `GenerateAssemblyVersionInfo` and `GetBuildVersion` then you could rewrite that property. But that strikes me as making your car float instead of building a boat.

Have you considered letting NB.GV do its thing? :wink:

Reply:
The project system is using NB.GV for this exact purpose. I was just trying to avoid adding dependencies for such a small thing, if it supports it, happy to do so if you prefer.

Reply:
Updated to use a SG